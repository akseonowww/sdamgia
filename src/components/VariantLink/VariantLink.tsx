import React, { FC, ReactNode } from 'react'
import cx from 'classnames'

import Link from 'components/Link'
import './VariantLink.scss'

export interface IVariantLinkProps {
  className?: string
  to: string
  title?: string
  children: ReactNode
  'data-wordClassName'?: string
}

const VariantLink: FC<IVariantLinkProps> = props => {
  const { className, to, children, title } = props
  const dataWordClassName = props['data-wordClassName']

  return (
    <Link className={cx('VariantLink', className)} to={to} title={title}>
      {!Number.isNaN(Number(children)) && dataWordClassName && (
        <span className={dataWordClassName}>Вариант&nbsp;</span>
      )}
      {children}
    </Link>
  )
}

export default VariantLink
