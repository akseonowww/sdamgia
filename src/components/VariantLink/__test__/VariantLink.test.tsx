import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import VariantLink, { IVariantLinkProps } from '../VariantLink'

jest.mock('react-router-dom', () => ({
  Link: ({ className, to, title, children }: IVariantLinkProps) => (
    <div className={className} data-to={to} title={title}>
      {children}
    </div>
  ),
}))

describe('VariantLink', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with required and optional props', () => {
    act(() => {
      render(<VariantLink to="/test?id=1">1</VariantLink>, container)
    })
    expect(container && container.textContent).toBe('1')

    act(() => {
      render(
        <VariantLink
          className="LarinVariants-Link"
          data-wordClassName="LarinVariants-Word"
          to="/test?id=1"
          title="Larin Variant #1"
        >
          1
          <div className="LarinVariants-Hint" title="Помощь">
            ?
          </div>
        </VariantLink>,
        container
      )
    })
    expect(container && container.textContent).toBe('1?')
  })

  it('should render a number', () => {
    act(() => {
      render(<VariantLink to="/test?id=1">1</VariantLink>, container)
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <VariantLink
          className="LarinVariants-Link"
          data-wordClassName="LarinVariants-Word"
          to="/test?id=1"
          title="Larin Variant #1"
        >
          1
          <div className="LarinVariants-Hint" title="Помощь">
            ?
          </div>
        </VariantLink>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
