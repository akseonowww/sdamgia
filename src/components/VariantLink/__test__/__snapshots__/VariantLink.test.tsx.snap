// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantLink should render a number 1`] = `"<div class=\\"Link VariantLink\\" data-to=\\"/test?id=1\\">1</div>"`;

exports[`VariantLink should render a number 2`] = `
"<div class=\\"Link VariantLink LarinVariants-Link\\" data-to=\\"/test?id=1\\" title=\\"Larin Variant #1\\">1<div class=\\"LarinVariants-Hint\\" title=\\"Помощь\\">?</div>
</div>"
`;

exports[`VariantLink should render a number 3`] = `"<div class=\\"Link VariantLink LarinVariants-Link\\" data-to=\\"/test?id=1\\" title=\\"Larin Variant #1\\"><span class=\\"LarinVariants-Word\\">Вариант&nbsp;</span>1</div>"`;
